### PR TITLE
Strip doc_string from exported ONNX models

### PR DESCRIPTION
**Pull request by lara-hdr**
Strip the doc_string by default from the exported ONNX models (this string has the stack trace and information about the local repos and folders, which can be confidential).

The users can still generate the doc_string by specifying add_doc_string=True in torch.onnx.export().

### CODE REVIEW

**Comment by ezyang** [comment]
The patch looks reasonable. It is technically BC-breaking because the default behavior changes; maybe that's OK. It's a little weird that you flip the meaning of the boolean between the frontend and backend.

**Comment by lara-hdr** [comment]
@ezyang, I set the default to strip the doc string as it is more confidential, and the stack trace does not add much value to the generated model.

The backend already uses "strip_doc_string" which I didn't want to change. From the front end side, it seemed to me that add_doc_string would make more sense from the user's point of view (especially if the default is generating the model without the doc_string), which is why I flip the meaning of the boolean. But I can change it to use strip_doc_string everywhere (or add_doc_string everywhere).

**Comment by lara-hdr** [comment]
@houseroad any update on this?

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -41,7 +41,8 @@ def set_training(model, mode):
 
 def export(model, args, f, export_params=True, verbose=False, training=False,
            input_names=None, output_names=None, aten=False, export_raw_ir=False,
-           operator_export_type=None, opset_version=None, _retain_param_name=True):
+           operator_export_type=None, opset_version=None, add_doc_string=False,
```
Comment (houseroad) [inline torch/onnx/utils.py]: add the add_doc_string at the end

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -105,7 +109,7 @@ def export(model, args, f, export_params=True, verbose=False, training=False,
             operator_export_type = OperatorExportTypes.ONNX
     _export(model, args, f, export_params, verbose, training, input_names, output_names,
             operator_export_type=operator_export_type, opset_version=opset_version,
-            _retain_param_name=_retain_param_name)
+            add_doc_string=add_doc_string, _retain_param_name=_retain_param_name)
```
Comment (houseroad) [inline torch/onnx/utils.py]: ditto

File: torch/onnx/utils.py [hunk 3]
```diff
@@ -307,7 +311,7 @@ def _export_to_pretty_string(model, args, f, export_params=True, verbose=False,
 def _export(model, args, f, export_params=True, verbose=False, training=False,
             input_names=None, output_names=None, operator_export_type=OperatorExportTypes.ONNX,
             export_type=ExportTypes.PROTOBUF_FILE, example_outputs=None, propagate=False,
-            opset_version=None, _retain_param_name=False):
+            opset_version=None, add_doc_string=False, _retain_param_name=False):
```
Comment (houseroad) [inline torch/onnx/utils.py]: ditto

File: torch/csrc/jit/export.cpp [hunk 4]
```diff
@@ -989,14 +989,15 @@ std::tuple<std::string, RawDataExportMap> export_onnx(
     const std::map<std::string, at::Tensor>& initializers,
     int64_t onnx_opset_version,
     bool defer_weight_export,
-    ::torch::onnx::OperatorExportTypes operator_export_type) {
+    ::torch::onnx::OperatorExportTypes operator_export_type,
+    bool strip_doc_string) {
```
Comment (houseroad) [inline torch/csrc/jit/export.cpp]: yeah, let's use strip_doc_string everywhere.

**Review by houseroad — COMMENTED** [review]
stripping doc by default should be fine. could you address my comments and resolve the conflicts?

**Review by houseroad — COMMENTED** [review]
Please also add a test case here: https://github.com/pytorch/pytorch/blob/master/test/onnx/test_utility_funs.py#L10

**Review by houseroad — APPROVED** [review]
Looks good. Thanks!

**Comment by houseroad** [comment]
rebase and resolve the conflicts?

**Comment by lara-hdr** [comment]
@houseroad, I resolved the conflicts, could we merge this?
Thank you!

**Pull request closed without merging**